crates/bevy_ui/src/flex/convert.rs

```diff
@@ -3,65 +3,113 @@ use crate::{
-        align_content: value.align_content.into(),
-        justify_content: value.justify_content.into(),
-        position: from_rect(scale_factor, value.position),
-        margin: from_rect(scale_factor, value.margin),
-        padding: from_rect(scale_factor, value.padding),
-        border: from_rect(scale_factor, value.border),
+        align_content: Some(value.align_content.into()),
+        justify_content: Some(value.justify_content.into()),
+        inset: from_rect(scale_factor, value.position),
+        margin: taffy::geometry::Rect {
+            left: margin(scale_factor, value.margin.left),
+            right: margin(scale_factor, value.margin.right),
+            top: margin(scale_factor, value.margin.top),
+            bottom: margin(scale_factor, value.margin.bottom),
+        },
+        padding: taffy::geometry::Rect {
+            left: length_percent(scale_factor, value.padding.left),
+            right: length_percent(scale_factor, value.padding.right),
+            top: length_percent(scale_factor, value.padding.top),
+            bottom: length_percent(scale_factor, value.padding.bottom),
+        },
+        border: taffy::geometry::Rect {
+            left: length_percent(scale_factor, value.border.left),
+            right: length_percent(scale_factor, value.border.right),
+            top: length_percent(scale_factor, value.border.top),
+            bottom: length_percent(scale_factor, value.border.bottom),
+        },
         flex_grow: value.flex_grow,
         flex_shrink: value.flex_shrink,
-        flex_basis: from_val(scale_factor, value.flex_basis),
-        size: from_val_size(scale_factor, value.size),
-        min_size: from_val_size(scale_factor, value.min_size),
-        max_size: from_val_size(scale_factor, value.max_size),
+        flex_basis: dimension(scale_factor, value.flex_basis),
+        size: from_size(scale_factor, value.size),
+        min_size: from_size(scale_factor, value.min_size),
+        max_size: from_size(scale_factor, value.max_size),
         aspect_ratio: value.aspect_ratio,
-        gap: from_val_size(scale_factor, value.gap),
+        gap: taffy::geometry::Size {
+            width: length_percent(scale_factor, value.gap.width),
+            height: length_percent(scale_factor, value.gap.height),
+        },
+        justify_self: None,
     }
 }
 
-pub fn from_val(scale_factor: f64, val: Val) -> taffy::style::Dimension {
+fn dimension(scale_factor: f64, val: Val) -> taffy::style::Dimension {
     match val {
-        Val::Auto => taffy::style::Dimension::Auto,
+        Val::Auto | Val::Undefined => taffy::style::Dimension::Auto,
         Val::Percent(value) => taffy::style::Dimension::Percent(value / 100.0),
         Val::Px(value) => taffy::style::Dimension::Points((scale_factor * value as f64) as f32),
-        Val::Undefined => taffy::style::Dimension::Undefined,
+    }
+}
+
+fn length_percent(scale_factor: f64, val: Val) -> taffy::style::LengthPercentage {
```

Comment:
These could also be `From`/`Into` implementations. Although this way is fine too.

Reply:
> These could also be `From`/`Into` implementations. Although this way is fine too.

I began some trait implementations but decided it's not worth doing until `Val::Undefined` is removed etc.